components/hab/src/command/pkg/delete.rs

```diff
@@ -0,0 +1,47 @@
+//! Delete a package from Builder.
+//!
+//! # Examples
+//!
+//! ```bash
+//! $ hab pkg delete acme/redis/2.0.7/2112010203120101 stable
+//! ```
+//! //! This will delete the acme package specified from Builder
+//! //! if certain conditions apply - for example, the package is not
+//! //! in the stable channel, and does not have any other packages that
+//! //! depend on it.
+//!
+//! //! Note: This command does not remove the package from disk
+
+use crate::{api_client::Client,
+            common::ui::{Status,
+                         UIWriter,
+                         UI},
+            hcore::package::PackageIdent};
+
+use crate::{error::{Error,
+                    Result},
+            PRODUCT,
+            VERSION};
+
+/// Delete a package from Builder.
+///
+/// # Failures
+///
+/// * Fails if it cannot find the specified package in Builder
+pub fn start(ui: &mut UI, bldr_url: &str, ident: &PackageIdent, token: &str) -> Result<()> {
+    let api_client = Client::new(bldr_url, PRODUCT, VERSION, None)?;
+
+    ui.begin(format!("Deleting {} from Builder", ident))?;
+
+    match api_client.delete_package(ident, token) {
+        Ok(_) => (),
+        Err(e) => {
+            println!("Failed to delete '{}': {:?}", ident, e);
+            return Err(Error::from(e));
+        }
+    }
```

Comment:
Why not
```rust
    if let Err(e) = api_client.delete_package(ident, token) {
        println!("Failed to delete '{}': {:?}", ident, e);
        return Err(Error::from(e));
    }
```
?

Reply:
Surprised clippy didn't call that out... will update